framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4622,447 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <p>{@code type} should be captured if all of the following are true:
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} are both declared types.
+   *   <li>{@code type} does not have an uninferred type argument and its underlying type is not
+   *       raw.
+   *   <li>{@code type} has a wildcard as a type argument and {@code typeMirror} has a captured type
+   *       variable as the corresponding type argument.
+   * </ul>
+   *
+   * @param type annotated type that might need to be captured
```

Comment:
Neither of these are true.   This method is checking if `type` should be captured.  If `type` has a wildcard as a type argument (and allow the other conditions are met), then it should be captured.  I'm not sure how to clarify the documentation.

Reply:
In certain parts of the code, there are application invariants about what values are passed to calls.  I was wondering if there are any here.

For example, in this method there is a specific relationship between `type` and `typeMirror`.  Since `typeMirror` is a capture, then the underlying type of `type` was capturable.  I'm wondering what that implies about `type`.  For example, I think it is impossible for `type` to be `int`, because its underlying type does not have a capture and because of where `applyCaptureConversion` is called.  (Maybe, even if that fact is true, it is an accident of the current call patterns and is not part of the specification or contract of this method.)

Or, maybe your response here is saying those two facts:
 * `typeMirror` might be a *degenerate* capture -- that is, it might be the underlying type itself. 
 * this method might also perform a degenerate capture.

I hope all this clarifies my question.  Please ask again if not.